src/psyclone/tests/dynamo0p3_test.py

```diff
@@ -290,9 +292,10 @@ def test_ad_field_init_wrong_iteration_space():
         LFRicArgDescriptor(
             field_arg, metadata.iterates_over)._init_field(
                 field_arg, "ncolours")
-    assert ("LFRicArgDescriptor._init_field(): Invalid iteration "
-            "space 'ncolours' in the kernel metadata (expected "
-            "one of ['cells', 'dofs'])." in str(excinfo.value))
+    assert ("LFRicArgDescriptor._init_field(): Invalid operates-on "
```

Comment:
As commented in `LFRicArgDescriptor`, I suggest removing the routine name and replacing `operates-on` with `operatates_on`.